config/internal/configsource/retrieved.go

```diff
@@ -0,0 +1,42 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configsource
+
+import (
+	"go.opentelemetry.io/collector/config/experimental/configsource"
+)
+
+type retrieved struct {
```

Comment:
If this is for testing only perhaps move to _test.go file, otherwise it is a bit confusing. I was thinking it is somehow used in non-test go but could not find any usage.
Also existence of both `retrieved` and `Retrieved` is a bit confusing. Perhaps `retrieved` need a different name? Otherwise it sound it is the canonical of implementation of the interface, which I think it is not, right, it is just a helper for tests?

Reply:
Only test implementation will be used.